exporters/elasticsearch-exporter/src/test/java/io/camunda/zeebe/exporter/ElasticsearchExporterConfigurationTest.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.exporter;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import io.camunda.zeebe.exporter.api.ExporterException;
+import io.camunda.zeebe.exporter.api.context.Configuration;
+import io.camunda.zeebe.exporter.api.context.Context;
+import io.camunda.zeebe.exporter.api.context.Context.RecordFilter;
+import io.camunda.zeebe.protocol.record.RecordType;
+import io.camunda.zeebe.protocol.record.ValueType;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.slf4j.Logger;
+
+final class ElasticsearchExporterConfigurationTest {
+
+  @Test
+  void shouldConfigureValueTypeFilter() {
+    // given
+    final var exporter = new ElasticsearchExporter();
+
+    final var enabledValueTypes =
+        Stream.of(ValueType.DEPLOYMENT, ValueType.PROCESS, ValueType.ERROR, ValueType.INCIDENT);
+    final var disabledValueTypes =
+        Stream.of(
+            ValueType.VARIABLE_DOCUMENT,
+            ValueType.PROCESS_INSTANCE_CREATION,
+            ValueType.PROCESS_MESSAGE_SUBSCRIPTION);
+
+    final var config = new ElasticsearchExporterConfiguration();
+    config.index.deployment = true;
+    config.index.process = true;
+    config.index.error = true;
+    config.index.incident = true;
+    config.index.variableDocument = false;
+    config.index.processInstanceCreation = false;
+    config.index.messageSubscription = false;
+
+    final var configuration = mock(Configuration.class);
+    when(configuration.instantiate(eq(ElasticsearchExporterConfiguration.class)))
+        .thenReturn(config);
+
+    final var capturedFilter = ArgumentCaptor.forClass(RecordFilter.class);
+    final var context = mock(Context.class);
+    when(context.getLogger()).thenReturn(mock(Logger.class));
+    when(context.getConfiguration()).thenReturn(configuration);
+
+    // when
+    exporter.configure(context);
+    verify(context).setFilter(capturedFilter.capture());
+    final var filter = capturedFilter.getValue();
+
+    // then
+    assertThat(enabledValueTypes).allMatch(filter::acceptValue);
+    assertThat(disabledValueTypes).noneMatch(filter::acceptValue);
```

Comment:
:wrench: An alternative would be to extract the `ElasticsearchRecordFilter`, implemented `equals/hashCode`, and construct here an expected filter and compare the actual with the expected. That way we can reuse how the filter is created and we're not testing its behavior. Of course, we need a unit test that the filter works as expected...which I guess is what this test is doing? But then both tests have narrower scopes, and the filter test has a much simpler set up. Just a suggestion.

Reply:
That's a good idea. I'd like to this once the Java 17 upgrade is done because most(all?) of the configuration classes could be records which will give us the equals/hashCode for free.